src/node/hooks/express/errorhandling.js

```diff
@@ -2,17 +2,22 @@
 
 const stats = require('../../stats');
 
-exports.expressCreateServer = (hook_name, args, cb) => {
-  exports.app = args.app;
-
-  // Handle errors
+/**
+ * Express already comes with an error handler that is attached
+ * as the last middleware. Within all routes it's possible to call
+ * `next(err)` where `err` is an Error object. You can specify
+ * `statusCode` and `statusMessage`.
+ * For more details see "The default error handler" section on
+ * https://expressjs.com/en/guide/error-handling.html
+ *
+ * This method is only used for metrics
+ *
+ */
+exports.expressCreateServer = (hookName, args, cb) => {
   args.app.use((err, req, res, next) => {
-    // if an error occurs Connect will pass it down
-    // through these "error-handling" middleware
-    // allowing you to respond however you like
-    res.status(500).send({error: 'Sorry, something bad happened!'});
-    console.error(err.stack ? err.stack : err.toString());
-    stats.meter('http500').mark();
+    const status = err.statusCode || err.status;
+    stats.meter(`http${status}`).mark();
+    next(err);
```

Comment:
I'm a bit concerned about the `status` variable being unset/wrong. How about this instead:

```suggestion
    next(err);
    // Get the status code after next() has had a chance to update it.
    stats.meter(`http${res.statusCode || 500}`).mark();
```

Reply:
In my test (I put `throw new Error('bad bad');` right after `args.app.get('/', (req, res) => {` in specialpages.js) I get a response status code of 200 - which feels strange. (The res object is probably prepared in advance, but I thought it was enough to only throw the error without explicitly setting the error code of the response.)

I'm pretty sure I tested it in December but I probably always set a status myself, so I did not notice that it also can be undefined. (like: `err=new Error('bad bad');err.status=400;return next(err)`)

Right now we don't have any further error handler, so `next(err)` triggers express' default error handler (which prints something on console.error)

I think it's best if we had some test coverage here, but we would probably need to implement some kind of throwing route handler. What do you think if instead of using `res.statusCode` we use `err.statusCode || err.status || 500`? I'm not sure where the 200 I see is coming from (my wget gets a 500 though), but I think we should expect the err object to have the proper status code

Reply:
> In my test ... I get a response status code of 200 - which feels strange.

That's unfortunate. :slightly_frowning_face:

Reply:
I found a reliable way to get the HTTP status code and replaced your commit with the new approach. Please take a look.